File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/LivrusApp.pas
unit LivrusApp;

interface

uses
  System.Drawing, wx;

type
  TLivrusApp = class(wx.App)
  public
    constructor Create;
    Function OnInit(): Boolean; override;
  end;

implementation

uses Principal, Splash, Resource, RotinasGlobais, ConsultasSQL;

constructor TLivrusApp.Create;
begin
  inherited Create;
end;

Function TLivrusApp.OnInit(): Boolean;
var bret: boolean;
begin
  Res := TResource.create;
  Rotinas := TRotinasGlobais.create;
  ConsSQL := TConsultasSQL.create;
  frmPrincipal := TfrmPrincipal.Create();

  if not Rotinas.ConectaBanco(
    'File Name=C:\LivrusVirtus\bancos\access\Access2k_Livrus.udl') then
    bret := false
  else begin
    bret := true;
    frmSplash := TfrmSplash.create(nil);
    frmSplash.show(true);
  end;
  result := bret;
end;

end.

File: LivrusVirtus/wxWindows/wxJava/src/Livrus/LivrusApp.java
package Livrus;

import org.wxwindows.wxApp;
import org.wxwindows.wxPoint;
import org.wxwindows.wxSize;

public class LivrusApp extends wxApp{
  public boolean OnInit(){
    wxInitAllImageHandlers();

    RotinasGlobais.Rotinas = new RotinasGlobais();
    ConsultasSQL.ConsSQL = new ConsultasSQL();
    frmPrincipal.fPrincipal = new frmPrincipal();

    if (RotinasGlobais.Rotinas.ConectaBanco()==false)
      return false;

    frmSplash.fSplash = new frmSplash();
    frmSplash.fSplash.Show(true);

    //frmPrincipal.fPrincipal.Show(true);
    return true;
  }

  public static void main(String args[]){
    wxApp aplicativo = new LivrusApp();
    aplicativo.MainLoop();
  }
}
